src/awkward/_nplikes/placeholder.py

```diff
@@ -22,16 +22,18 @@ def __init__(
         nplike: NumpyLike,
         shape: tuple[ShapeItem, ...],
         dtype: DType,
-        field_path: tuple[str, ...] = (),
+        field_path: tuple = (),
     ):
         self._nplike = nplike
         self._shape = shape
         self._dtype = np.dtype(dtype)
+        if not isinstance(field_path, tuple):
+            raise TypeError(f"field_path must be a tuple, not {field_path}")
         self._field_path = field_path
 
     @property
     def field_path(self) -> str:
-        return ".".join(self._field_path)
+        return ".".join(map(str, self._field_path))
```

Comment:
Given that this is an internal type, I'd be more in favour of fixing the call-sites, or adding an assert in the constructor, to ensure that there are not Nones here?

Reply:
That's a good argument. It will be used likely only once in dask-awkward, and I can make sure that it will be correct there. I'll close this PR then :+1: